### ISSUE TITLE
[Research Topic] Differentiable Resist Modeling via Physics-Informed Neural Networks (PINN)

### ISSUE THREAD

**Issue by telleroutlook** [OpenLithoHub/OpenLithoHub#5]
# Motivation

The current resist modeling in OpenLithoHub uses a hard threshold (`apply_resist_threshold` in `src/openlithohub/_utils/forward_model.py`), which is differentiable only via a smoothed sigmoid surrogate (`_utils/resist_model.py`). This is fine for proof-of-concept ILT, but it ignores the underlying physics:

- **Resist diffusion** (acid blur in chemically-amplified resists)
- **Develop kinetics** — explicitly the **Mack model** (and the **enhanced Mack** variant for low-CD / high-NA)
- **Photoacid generator (PAG) statistics** (especially relevant for EUV)
- **Bake / develop coupling** (PEB temperature → diffusion length)

A **Physics-Informed Neural Network (PINN)** can in principle absorb partial-differential constraints (e.g., the reaction–diffusion PDE governing acid concentration during PEB) into the loss function, while remaining fully auto-differentiable end-to-end. That makes it a natural fit as a drop-in replacement for the current threshold model in any `LithographyModel.predict()` call.

## The PDE we want learned

Concretely the post-exposure bake (PEB) acid concentration `H(x, y, t)` and quencher `Q(x, y, t)` evolve as a coupled reaction–diffusion system:

```
∂H/∂t = D_H · ∇²H − k_q · H · Q
∂Q/∂t = D_Q · ∇²Q − k_q · H · Q
```

with initial condition `H(x, y, 0) ∝ aerial_intensity(x, y)`. The development step is then a Mack-style rate `r(H) = r_max · (a + 1)·(1 − H)^n / (a + (1 − H)^n)` integrated to the threshold-clear time.

A PINN should learn `H, Q` (or directly the developed-resist field) under the residual of those PDEs at random collocation points.

## Hard requirement: analytic monotonicity

The output **must be monotone in dose** (more light → more clearing) over the physical operating range, with no gradient ringing. A non-monotone surrogate destabilises any downstream OPC loop that uses the resist model as part of its objective — the whole point of moving past the sigmoid surrogate is to add physics, not to add new failure modes.

# What we want

A new module — sketch path: `src/openlithohub/models/pinn_resist.py` — implementing a class roughly like:

```python
from openlithohub.models.base import LithographyModel, PredictionResult

class PINNResistModel(LithographyModel):
    NAME = "pinn-resist"
    SUPPORTS_CURVILINEAR = True

    def __init__(
        self,
        *,
        peb_diffusion_length_nm: float = 25.0,
        mack_n: float = 5.0,
        mack_a: float = 0.05,
        ...
    ): ...

    def predict(self, design: torch.Tensor, **kwargs) -> PredictionResult:
        # design: aerial image tensor (H,W) or (B,1,H,W) in [0, 1]
        # returns: PredictionResult with `mask` field as developed resist
        # ...
```

Concretely we'd love to see:

1. A small UNet (or FNO / PINN architecture of your choice) that maps `aerial_image → developed_resist`.
2. A loss that combines:
   - **Data fit** against ground-truth resist contours (use any of our datasets — `OpenLithoHub/litho-tiny` works as a trivial smoke test).
   - **Physics residual** of the reaction-diffusion PDE evaluated at random collocation points.
   - **Monotonicity penalty** — e.g. `relu(−∂resist/∂dose)` integrated over the patch.
3. A short Colab / notebook reproducing the training curve.

# Interface contract

> **You only need to subclass `LithographyModel` (`src/openlithohub/models/base.py`) and ensure `predict(design)` returns a `PredictionResult`.** EPE computation, PV-band, MRC checking, leaderboard registration, and GDS export are handled by us.

The `forward(mask_tensor) → resist_tensor` signature mentioned in the strategic plan is honored exactly via `predict(design)` for OpenLithoHub. See `src/openlithohub/models/levelset_ilt.py` for a worked reference implementation of the same interface.

# Why this is a great research target

- **AI for Science + PDE solvers** — PINN is one of the most cited families in the SciML literature.
- **Lithography is a perfect proving ground** — small spatial domain, well-understood physics, plenty of public datasets.
- **Conferences that love this topic**: IEEE TCAD, ICCAD, NeurIPS (AI4Science track), SPIE Advanced Lithography, DAC.

# Recommended starting points

- Mack, "Fundamental Principles of Optical Lithography" — Ch. 7-9 on resist modeling.
- Raissi et al. (2019), "Physics-informed neural networks" — the canonical PINN paper.
- Lin et al. (2024), "Differentiable Resist Modeling for ILT" — recent EDA-side work.

# Win-win promise

> *This is a highly open research problem perfect for PhD students or ML researchers looking to publish at EDA/AI conferences (ICCAD, DAC, NeurIPS, SPIE). If you decide to tackle this, the core maintainers will provide full support on the OpenLithoHub framework side. We encourage you to publish papers based on this, and we will feature your merged implementation and paper on our official website's [Algorithms Showcase](https://openlithohub.com/showcase) section.*

---

Labels: `research-topic`, `phd-friendly`, `help wanted`, `call-for-paper`, `physics-informed`, `pinn`

---

## 🎯 What we expect from contributors (Your Scope)

**You do NOT need real fab calibration data to contribute here.** This is an *algorithmic* problem, not a *calibration* problem.

Your scope: build a differentiable PINN solver (the reaction–diffusion PDE above, in PyTorch) trained on **synthetic resist parameters from published literature** — Mack's CAR model coefficients, EUV stochastic blur σ from public SPIE / De Bisschop work, PEB diffusion lengths from textbook ranges. Every parameter you need is already in the open literature; none of it requires an NDA or a fab tool.

The acceptance bar for a merged PR:

- [ ] PDE residual loss converges on the synthetic test fixtures we ship in `tests/fixtures/`
- [ ] Forward pass is `torch.compile`-clean and runs on both CPU and CUDA
- [ ] Numerical gradient check passes (`torch.autograd.gradcheck`) at a representative collocation point
- [ ] **Analytic monotonicity in dose** is enforced (the hard requirement above) — verified by a unit test
- [ ] One short notebook showing the trained model recovers a known analytic solution (1D step or Gaussian source) within a documented tolerance

**What's explicitly out of scope** (and will be handled by industrial end-users with their own calibration loops): matching real fab CD-SEM measurements, vendor-specific resist-stack tuning, NDA'd process windows. The synthetic→real bridge is *not* part of this PR.

If your PR clears the boxes above, it lands. We'd rather merge a clean, well-tested solver on synthetic data than wait for a fab partnership that may never come — once the framework is in, an industrial user with calibration data can fine-tune in their own private fork.
